100of100Service/src/main/java/dev/coms4156/project/Department.java

```diff
@@ -1,25 +1,64 @@
 package dev.coms4156.project;
 
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * This class represents a department in the organization.
  * Designed under the Composite Design Pattern.
  */
 public class Department extends OrganizationComposite {
   private Employee head;
+  private final List<Employee> employees;
 
   /**
    * Constructs a department with the given ID and name.
-   * @param db the HR database facade that manages the department
-   * @param id the ID of the department
+   *
+   * @param db   the HR database facade that manages the department
+   * @param id   the ID of the department
    * @param name the name of the department
    */
-  public Department(HRDatabaseFacade db, long id, String name) {
+  public Department(HRDatabaseFacade db, int id, String name, List<Employee> employees) {
     super(db, id, name);
+    this.typeName = "Department";
+    this.employees = new ArrayList<>();
     this.head = null;
   }
 
+  /**
+   * Adds an employee to the department.
+   *
+   * @param employee the employee to be added
+   * @return true if the employee is added successfully, false otherwise
+   */
+  public boolean addEmployee(Employee employee) {
+    this.employees.add(employee); // Add to employee list
+    return this.add(employee);
+  }
+
+  /**
+   * Removes an employee from the department.
+   *
+   * @param employee the employee to be removed
+   * @return true if the employee is removed successfully, false otherwise
+   */
+  public boolean removeEmployee(Employee employee) {
+    this.employees.remove(employee); // Remove from employee list
+    return this.remove(employee);
+  }
+
+  /**
+   * Returns the list of employees in the department.
+   *
+   * @return the list of employees
+   */
+  public List<Employee> getEmployees() {
+    return new ArrayList<>(this.employees); // Return a copy of the employee list
```

Comment:
Should we just use `.clone()`? And, why do we especially want a copy here? 
Btw, is this still a shallow copy or a deep copy?